.github/workflows/e2e.yml

```diff
@@ -0,0 +1,118 @@
+        uses: actions/checkout@v4
+        with:
+          repository: kyma-project/eventing-manager
+          path: eventing-manager
+
+      - name: Checkout NM repository
+        uses: actions/checkout@v4
+        with:
+          repository: kyma-project/nats-manager
+          path: nats-manager
+
+      - name: Cache binaries
+        id: cache-binaries
+        uses: actions/cache@v3
+        with:
+          path: bin
+          key: ${{ runner.os }}-bin
+
+      - name: Install k3d tools
+        run: |
+          make -C eventing-manager/hack/ci/ install-k3d-tools
+
+      - name: Install Kyma CLI & setup k3d cluster using kyma CLI
+        run: |
+          make -C eventing-manager kyma
+          make -C main/hack/ci/ create-k3d
+          kubectl version
+          kubectl cluster-info
+
+      - name: Create Namespace
+        run: |
+          kubectl create ns kyma-system
+
+      - name: Setup NATS Manager
+        run: |
+          make -C nats-manager install
+
+      - name: Deploy NATS Manager
+        run: |
+          make -C nats-manager deploy IMG=$NATS_MANAGER_IMAGE
+
+      - name: Create NATS CR and resources and check configuration
+        run: |
+          make -C nats-manager e2e-setup
+
+      - name: Setup Eventing Manager
+        run: |
+          make -C eventing-manager install
+
+      - name: Deploy Eventing Manager
+        run: |
+          make -C eventing-manager deploy IMG=$EVENTING_MANAGER_IMAGE
+
+      - name: Wait for build job to succeed
+        uses: kyma-project/wait-for-commit-status-action@2b3ffe09af8b6f40e1213d5fb7f91a7bd41ffb20
+        with:
+          context: "pull-eventing-publisher-proxy-build"
+          commit_ref: "${{ github.event.pull_request.head.sha }}" # Note: 'github.event.pull_request.head.sha' is not same as 'github.sha' on pull requests.
+          timeout: 600000 # 10 minutes in milliseconds
+          # The check interval is kept long otherwise it will exhaust the GitHub rate limit (More info: https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting)
+          check_interval: 60000 # 1 minute in milliseconds
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          GITHUB_OWNER: "${{ github.repository_owner }}"
+          GITHUB_REPO: "eventing-publisher-proxy"
+
+      - name: Setup Eventing Manager with new EPP image
+        run: |
+          kubectl patch deployment eventing-manager -n kyma-system --type='json' -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/env/8/value", "value": "'$EPP_IMAGE'"}]'
+
+      - name: Create eventing CR and resources and check configuration
+        run: |
+          make -C eventing-manager e2e-setup
+
+      - name: Setup subscriptions for testing
+        run: |
+          make -C eventing-manager e2e-eventing-setup
+
+      - name: Test eventing
+        run: |
+          make -C eventing-manager e2e-eventing
```

Comment:
Should we also run the test cleanup step?

Reply:
Yes!

Reply:
Done